Parse/src/main/java/com/parse/ParseFileController.java

```diff
@@ -56,6 +56,13 @@ public File getCacheFile(ParseFile.State state) {
     return new File(cachePath, state.name());
   }
 
+  /* package for tests */ File getTempFile(ParseFile.State state) {
+    if (state.url() == null) {
+      return null;
+    }
+    return new File(cachePath, ParseDigestUtils.md5(state.url()));
```

Comment:
Can we guarantee the url of a ParseFile should always be different? If so, why we will have clash. But I will make the change, since UUID is definitely much safer. Just curious about the question.
For clash, I think 
`  public File getCacheFile(ParseFile.State state) {
    return new File(cachePath, state.name());
  }`
will have clash, since for ParseFile create without a given name, `state.name` will always be file. Is that correct？